node_modules/oae-mixpanel/tests/test-content.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var assert = require('assert');
+
+var RestAPI = require('oae-rest');
+var TestsUtil = require('oae-tests');
+
+var MixPanelTestUtil = require('oae-mixpanel/lib/test/util');
+
+describe('MixPanel - Content', function() {
+
+    // Rest context that can be used every time we need to make a request as a global admin
+    var globalAdminRestContext = null;
+    // Rest context that can be used every time we need to make a request as an anonymous user on the global admin tenant
+    var anonymousGlobalRestContext = null;
+    // Rest context that can be used every time we need to make a request as a cam tenant admin
+    var camAdminRestContext = null;
+    // Rest context that can be used every time we need to make a request as an anonymous user
+    var anonymousCamRestContext = null;
+
+    /**
+     * Initializes the REST contexts
+     */
+    before(function() {
+        // Fill up the global admin rest context
+        globalAdminRestContext = TestsUtil.createGlobalAdminRestContext();
+        // Fill up the anonymous global rest context
+        anonymousGlobalRestContext = TestsUtil.createGlobalRestContext();
+        // Fill up the cam admin rest context
+        camAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.cam.host);
+        // Fill up anonymous rest context
+        anonymousCamRestContext = TestsUtil.createTenantRestContext(global.oaeTests.tenants.cam.host);
+    });
+
+    /**
+     * Enable MixPanel before each test
+     */
+    beforeEach(function() {
+        require('../../../config').config.mixpanel.enabled = true;
+    });
+
+    /**
+     * Disable MixPanel when all the tests are ran
+     */
+    after(function() {
+        require('../../../config').config.mixpanel.enabled = false;
```

Comment:
When switching to the application initialization method of handling config, you could instead put functions on the MixPanelAPI that allows you to enable and disabled it.